### PR TITLE
[ENH] migrate MCDCNN classifier, regressor, network from `sktime-dl`

### LINKED ISSUE #3295: [ENH] Adding MCDNN DL model for Classification
**Is your feature request related to a problem? Please describe.**
Migrating MCDNN  model from sktime-dl to sktime for classification.

### PR DESCRIPTION
**Pull request by achieveordie**
#### Reference Issues/PRs

Fixes #3295
Part of #3351

#### What does this implement/fix? Explain your changes.

Add `MCDCNNClassifier`, `MCDCNNRegressor` and `MCDCNNNetwork`

#### Does your contribution introduce a new dependency? If yes, which one?

None

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [ ] Optionally, I've added myself and possibly others to the [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) file - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [x] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [x] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [x] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Comment by achieveordie** [comment]
@fkiraly could you please review this too?

File: sktime/classification/deep_learning/mcdcnn.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
+# -*- coding: utf-8 -*-
+"""Multi Channel Deep Convolutional Neural Classifier (MCDCNN)."""
+
+__author__ = [
+    "James Large",
```
Comment (fkiraly) [inline sktime/classification/deep_learning/mcdcnn.py:5]: this should be GitHub ID, I think it's `JamesLarge`

File: sktime/regression/deep_learning/mcdcnn.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        ----------
+        X : an np.ndarray of shape = (n_instances, n_dimensions, series_length)
+            The testing input samples.
+
+        Returns
+        -------
+        output : array of shape = [n_instances,]
+            Representing the estimates for all instances in X.
+        """
+        self.check_is_fitted()
```
Comment (fkiraly) [inline sktime/regression/deep_learning/mcdcnn.py:234]: check_is_fitted is done already by `predict`

**Review by fkiraly — APPROVED** [review]
Thanks - left some non-blocking comments, but would appreciate a quick look.

**Comment by achieveordie** [comment]
Currently,  Ubuntu-3.11 is failing due to:
> FAILED sktime/forecasting/base/tests/test_fh.py::test_auto_arima - ValueError: Could not successfully fit a viable ARIMA model to input data.
See http://alkaline-ml.com/pmdarima/no-successful-model.html for more information on why this can happen.

Is this a known issue? @yarnabrina @fkiraly

**Comment by yarnabrina** [comment]
Not known to me, and cannot reproduce locally.

Command used:

> pytest -k "test_auto_arima" sktime/forecasting/base/tests/test_fh.py

Versions:

```pycon
>>> from sktime import show_versions; show_versions()

System:
    python: 3.11.3 (main, Apr 19 2023, 23:54:32) [GCC 11.2.0]
executable: /path/to/venv/bin/python
   machine: Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35

Python dependencies:
          pip: 22.3.1
       sktime: 0.19.1
      sklearn: 1.2.2
        numpy: 1.24.3
        scipy: 1.10.1
       pandas: 2.0.2
   matplotlib: None
       joblib: 1.2.0
  statsmodels: 0.14.0
        numba: None
     pmdarima: 2.0.3
      tsfresh: None
   tensorflow: None
tensorflow_probability: None
```

**Review by fkiraly — DISMISSED** [review]
Looks good, we should be able to merge.

**Pull request merged**